File: frontend/src/types/appointment.ts
export interface Appointment {
  id: number;
  doctorId: number | undefined;
  patientId: number | undefined;
  date: string;
  startTime: string;
  endTime: string;
}

export type TimeSlotsType = {
  startTime: string;
  endTime: string;
};

export type AppointmentAvailabilitiesType = {
  doctorId: number;
  doctorName: string;
  timeRanges: Array<TimeSlotsType>;
};

export type AppointmentCallProps = {
  doctorId: number | undefined;
  patientId: number | undefined;
  date: string;
  startTime: string;
  endTime: string;
};

File: clinic/src/main/java/com/oot/clinic/entities/Appointment.java
package com.oot.clinic.entities;

import jakarta.persistence.*;

import java.time.LocalDate;
import java.time.LocalTime;

@Entity
public class Appointment {
    @Id
    @GeneratedValue
    private Long id;
    @ManyToOne
    @JoinColumn(name = "doctor_id")
    private Doctor doctor;
    @ManyToOne
    @JoinColumn(name = "patient_id")
    private Patient patient;
    private LocalDate date;
    private LocalTime startTime;
    private LocalTime endTime;

    public Appointment(Doctor doctor, Patient patient, LocalDate date, LocalTime startTime, LocalTime endTime) {
        this.doctor = doctor;
        this.patient = patient;
        this.date = date;
        this.startTime = startTime;
        this.endTime = endTime;
    }

    public Appointment() {}

    // GETTERS AND SETTERS

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public LocalTime getStartTime() {
        return startTime;
    }

    public void setStartTime(LocalTime startTime) {
        this.startTime = startTime;
    }

    public LocalTime getEndTime() {
        return endTime;
    }

    public void setEndTime(LocalTime endTime) {
        this.endTime = endTime;
    }

    public Patient getPatient() {
        return patient;
    }

    public void setPatient(Patient patient) {
        this.patient = patient;
    }

    public Doctor getDoctor() {
        return doctor;
    }

    public void setDoctor(Doctor doctor) {
        this.doctor = doctor;
    }


    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }


}
